.azure-pipelines/build-template.yml

```diff
@@ -0,0 +1,70 @@
+parameters:
+- name: platform
+  type: string
+  values:
+  - broadcom
+  - mellanox
+- name: platform_short
+  type: string
+  values:
+  - brcm
+  - mlnx
+
+- name: cache_mode
+  type: string
+  values:
+  - wcache
+  - rcache
+  - cache
+
+- name: dbg_image
+  type: boolean
+
+- name: swi_image
+  type: boolean
+
+- name: raw_image
+  type: boolean
+
+- name: sync_rpc_image
+  type: boolean
+
+jobs:
+- job:
+  displayName: ${{ parameters.platform }}
+  timeoutInMinutes: 600
+  steps:
+  - template: cleanup.yml
+  - checkout: self
+    submodules: recursive
+    displayName: 'Checkout code'
+  - script: |
+       git submodule foreach --recursive git clean -xfdf
+       git submodule foreach --recursive git reset --hard
+       git submodule update --init --recursive
+    displayName: 'reset submodules'
+  - script: |
```

Comment:
How about to move the script from the template to the official.yml, only keep the most common part in the template.

Reply:
I think we will have official-broadcom.yml, official-mellanox.yml, and so on. 
If they use different build scripts, It is hard to trace build error.

Reply:
each plat form write their own script. this will make yml file more different with each other, like current Jenkinsfiles.